CDP4WebServices.API/Services/Operations/OperationProcessor.cs

```diff
@@ -1313,12 +1333,14 @@ private void ExecuteDeleteOperation(Guid deletedValueId, string type, DtoResolve
             var propertyService = this.ServiceProvider.MapToPersitableService(type);
 
             Thing containerInfo = null;
+
             if (!metaInfo.IsTopContainer)
             {
                 containerInfo = this.GetContainerInfo(compositeThing).Thing;
             }
 
-            var persistedThing = compositeThing;
+            var persistedThing = resolvedInfo.Thing ?? compositeThing;
```

Comment:
looking at the code i think the variable called `compositeThing` is no longer required. Lets do all of it based on what is stored in the database, meaning the `resolvedInfo.Thing`, that gives all the information we need and then there is less confusion about what the difference could be between resolvedInfo.Thing and compositeThing.

Reply:
Refactored ApplyDeleteOperation and ExecuteDeleteOperation